### PR TITLE
feat(handlers): add group DM function

### PR DESCRIPTION
**Pull request by itohatweb**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [ ] Code changes have been tested against the Discord API, or there are no code changes

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/api/handlers/channel.ts [hunk 1]
```diff
@@ -471,3 +471,23 @@ export async function isChannelSynced(channelID: string) {
       overwrite.deny !== permission.deny);
   });
 }
+
+/** Adds a recipient to a Group DM using the app granted access token from the `gdm.join` scope */
+export function groupDmAddRecipient(
```
Comment (ayntee) [inline src/api/handlers/channel.ts:476]: A much predictable and pleasant function name is required, something like `addGroupDmRecipient`.

File: src/api/handlers/channel.ts [hunk 2]
```diff
@@ -471,3 +471,23 @@ export async function isChannelSynced(channelID: string) {
...
+  nickname: string,
+) {
+  return RequestManager.put(
+    endpoints.GROUP_DM_RECIPIENTS(channelID, userID),
+    { access_token: token, nick: nickname },
+  );
+}
+
+/** Removes a recipient from a Group DM */
+export function groupDmRemoveRecipient(channelID: string, userID: string) {
```
Comment (ayntee) [inline src/api/handlers/channel.ts:489]: Similarly, a better function name is required, something like `removeGroupDmRecipient`.

File: src/api/handlers/channel.ts [hunk 3]
```diff
@@ -471,3 +471,23 @@ export async function isChannelSynced(channelID: string) {
       overwrite.deny !== permission.deny);
   });
 }
+
+/** Adds a recipient to a Group DM using the app granted access token from the `gdm.join` scope */
+export function addGroupDmRecipient(
+  channelID: string,
+  userID: string,
+  token: string,
```
Comment (ayntee) [inline src/api/handlers/channel.ts:479]:
```suggestion
  accessToken: string,
```
Much precise and informative. :smile:

File: src/api/handlers/channel.ts [hunk 4]
```diff
@@ -471,3 +471,23 @@ export async function isChannelSynced(channelID: string) {
...
+/** Adds a recipient to a Group DM using the app granted access token from the `gdm.join` scope */
+export function addGroupDmRecipient(
+  channelID: string,
+  userID: string,
+  accessToken: string,
+  nickname: string,
+) {
+  return RequestManager.put(
+    endpoints.GROUP_DM_RECIPIENTS(channelID, userID),
+    { access_token: token, nick: nickname },
```
Comment (ayntee) [inline src/api/handlers/channel.ts:484]:
```suggestion
    { access_token: accessToken, nick: nickname },
```

File: src/api/handlers/channel.ts [hunk 5]
```diff
@@ -471,3 +471,23 @@ export async function isChannelSynced(channelID: string) {
       overwrite.deny !== permission.deny);
   });
 }
+
+/** Adds a recipient to a Group DM using the app granted access token from the `gdm.join` scope */
```
Comment (ayntee) [inline src/api/handlers/channel.ts:475]:
```suggestion
/**
 * Adds a recipient to a Group DM using their access token
 * @param accessToken access token of a user that has granted your app the `gdm.join` scope
 */
```

File: src/api/handlers/channel.ts [hunk 6]
```diff
@@ -471,3 +471,26 @@ export async function isChannelSynced(channelID: string) {
...
+  accessToken: string,
+  nickname: string,
+) {
+  return RequestManager.put(
+    endpoints.GROUP_DM_RECIPIENTS(channelID, userID),
+    { access_token: accessToken, nick: nickname },
+  );
+}
+
+/** Removes a recipient from a Group DM */
```
Comment (ayntee) [inline src/api/handlers/channel.ts:491]:
```suggestion
/** Removes a recipient from a Group DM, */
```

**Review by ayntee — APPROVED** [review]
LGTM — thanks, @itohatweb!

**Comment by itohatweb** [comment]
closing this due to oauth2 not being implemented in rn.

**Pull request closed without merging**